src/App.js

```diff
@@ -15,22 +16,25 @@ class App extends Component {
       movies: [],
       search: '',
       status: DEFAULT_STATUS,
-      color: 'black'
+      color: 'black',
+      history: []
     }
     this.handleSearchChange = this.handleSearchChange.bind(this)
     this.handleSubmit = this.handleSubmit.bind(this)
     this.clearSearch = this.clearSearch.bind(this)
   }
 
   fetchMovies(search){
-    const { movies, status, color } = this.state
+    let { movies, status, color, history } = this.state
```

Comment:
Since you are using `this.state.history` below, you don't need to destructure it here.

And below you can declare it newly: `const history = [search, ...this.state.history]`

Reply:
thanks. done. Also adding new changes in this next commit.